### PR TITLE
TSM dev regression fixes

### PR DESCRIPTION
**Pull request by jwilder**
* WAL write errors started occurring after #4966 was merged
* Performance issue introduced with cursor `Scan` calling `Contains` when #4969 was merged.  This code needs to be optimized further as it's doing too many searches.  
* Rollover TSM cutoff was removed which causes TSM files to get rewritten too frequently and slows everything down.  This will be fixed with the revised compaction process.
* Lowered default batch size since internal buffers are sized for ~5000 point writes or less.  10000 causes lots of allocations and GC cycles.  This doubles write throughput.

### CODE REVIEW

**Comment by jwilder** [comment]
@otoolep @pauldix

File: tsdb/engine/tsm1/wal.go [hunk 1]
```diff
@@ -198,6 +207,8 @@ func (l *WAL) writeToLog(entry WALEntry) (int, error) {
 		return -1, fmt.Errorf("error rolling WAL segment: %v", err)
 	}
 
+	l.mu.RLock()
```
Comment (otoolep) [inline tsdb/engine/tsm1/wal.go]: I don't think I see anything in your commit message about this change?

File: tsdb/engine/tsm1/file_store.go [hunk 2]
```diff
@@ -280,11 +280,6 @@ func (f *FileStore) next(key string, t time.Time) ([]Value, error) {
 	defer f.mu.RUnlock()
 
 	for _, fd := range f.files {
-		// Can this file possibly contain this key and timestamp?
```
Comment (otoolep) [inline tsdb/engine/tsm1/file_store.go]: This wasn't needed because caller takes care of it?

File: tsdb/engine/tsm1/wal.go (hunk 1 above)
Comment (jwilder) [inline tsdb/engine/tsm1/wal.go]: See first PR description item.

File: tsdb/engine/tsm1/file_store.go (hunk 2 above)
Comment (jwilder) [inline tsdb/engine/tsm1/file_store.go]: It's actually faster to just do the real search and see if we have any values to search to see if we have values then search again to get the values.

Comment (otoolep) [inline tsdb/engine/tsm1/file_store.go]: Nice.

**Comment by otoolep** [comment]
+1

**Comment by pauldix** [comment]
Are you saying that having larger batches negatively impacts performance?

**Pull request merged**